ACE/ace/RAW_Socket.cpp

```diff
@@ -0,0 +1,414 @@
+#include "ace/RAW_Socket.h"
+
+#include "ace/ACE.h"
+#include "ace/Log_Category.h"
+#if defined (ACE_HAS_ALLOC_HOOKS)
+# include "ace/Malloc_Base.h"
+#endif /* ACE_HAS_ALLOC_HOOKS */
+
+// Included so users have access to ACE_RECVPKTINFO and ACE_RECVPKTINFO6 .
+#include "ace/OS_NS_sys_socket.h"
+
+#if defined (ACE_HAS_IPV6) && defined (ACE_WIN32)
+#include /**/ <iphlpapi.h>
+#endif
+
+
+#if !defined (__ACE_INLINE__)
+#  include "ace/RAW_Socket.inl"
+#endif /* __ACE_INLINE__ */
+
+ACE_BEGIN_VERSIONED_NAMESPACE_DECL
+
+ACE_ALLOC_HOOK_DEFINE (ACE_RAW_SOCKET)
+
+#if defined (ACE_HAS_4_4BSD_SENDMSG_RECVMSG) || defined ACE_WIN32
+  #define ACE_USE_MSG_CONTROL
```

Comment:
Don't indent # of  preprecessor lines, those should be first character, you can use spaces after #